scout/config.py

```diff
@@ -0,0 +1,217 @@
+        """
+
+        if not schema_data:
+            schema_data = self.schema_data
+        validate(input_data, schema_data)
+
+    def initialize_argparse(self, parser: argparse.ArgumentParser):  # noqa: F821
+        """Initialize argument parser with yaml argument, as this is not included in the schema
+
+        Args:
+            parser (argparse.ArgumentParser): Parser to which arguments are added
+        """
+        parser.add_argument(
+            "-y", "--yaml",
+            type=Path,
+            help=("Path to YAML configuration file, arguments passed directly to the command "
+                  "line will take priority over arguments in this file")
+            )
+
+    def set_config_args(self, cli_args: list[str] = []):
+        """If there is a config file provided, store data and validate
+
+        Args:
+            cli_args (list[str], optional): Command line arguments. Defaults to [].
+        """
+
+        self.config_args = {}
+        self.args = self.parser.parse_args(cli_args)
+        if self.args.yaml:
+            self.config_args = self._load_config(self.args.yaml)
+            self._validate(self.config_args, self.schema_data)
+
+    def parse_args(self) -> argparse.Namespace:  # noqa: F821
+        """Parse and store arguments from config file and the command line. CLI arguments take
+            precedence over config file arguments. Once parsed, arguments are checked for invalid
+            combinations not covered by the schema.
+
+        Returns:
+            argparse.Namespace: Arguments object to be used in ecm_prep.py or run.py
+        """
+
+        # Update args with yaml arguments
+        if self.config_args:
+            config_key_args = self.config_args.get(self.key, {})
+            self.args = self.update_args(self.args, config_key_args)
+
+        # Ensure command-line args take precedence
+        self.args = self.parser.parse_args(self.cli_args, namespace=self.args)
+
+        # Check for valid arguments
+        self.check_dependencies(self.args)
+
+        return self.args
+
+    def update_args(self,
+                    existing_args: argparse.NameSpace,  # noqa: F821
+                    new_args: dict) -> argparse.NameSpace:  # noqa: F821
+        """Update argparse arguments NameSpace with args dictionary
+
+        Args:
+            existing_args (argparse.NameSpace): Arguments object to be updated
+            new_args (dict): Arguments and argument values used to update
+
+        Returns:
+            argparse.NameSpace: Updated version of existing_args
+        """
+
+        for k, v in new_args.items():
+            if isinstance(v, dict):
+                self.update_args(existing_args, v)
+            else:
+                existing_args.__dict__.update({k: v})
+
+        return existing_args
+
+    def check_dependencies(self, args: argparse.NameSpace):  # noqa: F821
+        """Allow for custom checks that are not easily defined directly in the schema
```

Comment:
Can you add another sentence or two explaining why these custom checks have to be done here? (I'm aiming for some additional information so that another developer who sees this function will know when to add something here.)